### PR TITLE
Upgrade SBT to 0.13.9

### PR DESCRIPTION
**Pull request by smarter**
Review by @DarkDimius .

### CODE REVIEW

**Comment by smarter** [comment]
> ++ sbt -Ddotty.jenkins.build=yes -Dfile.encoding=UTF-8 -ivy /home/jenkins/workspace/dotty-master-validate-junit/ivy2 -Dsbt.global.base=/home/jenkins/.sbt/0.13 -sbt-dir /home/jenkins/.sbt/0.13 update compile test
Downloading sbt launcher for 0.13.9:
  From  http://typesafe.artifactoryonline.com/typesafe/ivy-releases/org.scala-sbt/sbt-launch/0.13.9/sbt-launch.jar
    To  /home/jenkins/.sbt/launchers/0.13.9/sbt-launch.jar
Download failed. Obtain the jar manually and place it at /home/jenkins/.sbt/launchers/0.13.9/sbt-launch.jar
Build step 'Execute shell' marked build as failure

Ping @adriaanm or someone who knows how Jenkins is configured?

**Comment by SethTisue** [comment]
I can look into this tomorrow, I've fixed this same issue elsewhere before

**Comment by SethTisue** [comment]
I can't self-assign because I'm not a committer. add me plz?

**Comment by DarkDimius** [comment]
@SethTisue, invitation sent

**Comment by SethTisue** [comment]
fixed — sorry for the delay

**Comment by smarter** [comment]
Thanks!

**Comment by SethTisue** [comment]
looks the scalastyle running is hitting https://github.com/scalastyle/scalastyle/issues/156, which has been fixed and the fix merged, but not published. see [@ahirreddy's workaround](https://github.com/scalastyle/scalastyle/issues/156#issuecomment-137229733)

**Comment by smarter** [comment]
I think I'll just keep scalastyle 0.6.0 for now.

**Comment by smarter** [comment]
Ah, I misunderstood the problem, didn't get that it was caused by sbt itself.

**Comment by smarter** [comment]
@DarkDimius: I got an `AbstractMethodError` solved by a rebuild again: https://scala-ci.typesafe.com/job/dotty-master-validate-partest/538/console

**Comment by smarter** [comment]
Ping @DarkDimius

**Comment by DarkDimius** [comment]
LGTM

**Comment by DarkDimius** [comment]
This PR broke my local sbt setup:

```
Downloading sbt launcher for 0.13.9:
  From  http://typesafe.artifactoryonline.com/typesafe/ivy-releases/org.scala-sbt/sbt-launch/0.13.9/sbt-launch.jar
    To  /Users/dark/.sbt/launchers/0.13.9/sbt-launch.jar
Download failed. Obtain the jar manually and place it at /Users/dark/.sbt/launchers/0.13.9/sbt-launch.jar
```

@SethTisue: could you please put sbt 0.13.9 to typesafe artifactory?

File: project/plugins.sbt [hunk 1]
```diff
@@ -3,7 +3,9 @@
...
 
 // Scala IDE project file generator
-addSbtPlugin("com.typesafe.sbteclipse" % "sbteclipse-plugin" % "2.2.0")
-
-addSbtPlugin("org.scalastyle" %% "scalastyle-sbt-plugin" % "0.6.0")
+addSbtPlugin("com.typesafe.sbteclipse" % "sbteclipse-plugin" % "4.0.0")
 
+// See https://github.com/scalastyle/scalastyle/issues/156#issuecomment-137229733
+addSbtPlugin("org.scalastyle" %% "scalastyle-sbt-plugin" % "0.7.0" excludeAll(
+  ExclusionRule(organization = "com.danieltrinh")))
```
Comment (DarkDimius) [inline project/plugins.sbt]: needs an empty line here. Otherwise SBT complains.

**Comment by smarter** [comment]
Typesafe artifactory does not exist anymore as far as I understand: https://www.typesafe.com/blog/migrating-repos-to-bintray Try using the latest SBT launcher?

File: project/plugins.sbt (hunk 1 above)
Comment (dwijnand) [inline project/plugins.sbt]: No longer needed as of 0.13.7 (https://github.com/sbt/sbt/issues/1606).

**Comment by dwijnand** [comment]
Yeah make sure you upgrade to the lastest sbt.

**Comment by SethTisue** [comment]
@smarter and @dwijnand are right — if you have something that isn't finding 0.13.9, that something is outdated and/or misconfigured

**Pull request merged**